boot/core/src/boot/core.clj

```diff
@@ -1313,3 +1313,19 @@
           (require 'boot.repl)
           (def ~pod-id (boot.repl/launch-nrepl '~opts)))
         #(pod/with-eval-in p (.close ~pod-id))))))
+
+(defn launch-socket-server [{:keys [bind port accept]}]
+  (let [opts {:host (or bind "127.0.0.1")
+              :port (or port 0)
+              :name (gensym "socket-server")
+              :accept (or accept 'clojure.core.server/repl)}]
+    (try
+      (require 'clojure.core.server)
+      (catch java.io.FileNotFoundException e
+        (throw (ex-info "Socket server requires Clojure version 1.8.0 or above"
+                        {:version (clojure-version)}
+                        e))))
+    (let [effective-port (-> ((resolve 'clojure.core.server/start-server) opts)
+                             (.getLocalPort))]
+      (doto (io/file ".socket-port") .deleteOnExit (spit effective-port))
+      (util/info "Socket server started on port %s on host %s.\n" effective-port (:host opts)))))
```

Comment:
Basically `boot.core` cannot be loaded in a pod because it has stateful stuff that we don't want to replicate. There may be a moment where you want to launch a socket server in a pod though which, with the function in `boot.core`, would require copying the implementation somewhere where it can be loaded by the pod.

For this reason I'm suggesting we put the function somewhere where it is also available in pods. That could be `boot.util` or `boot.pod`, probably there are more, but these two come to mind...